tools/vsextension/ProjectWizard/WizardImplementation.cs

```diff
@@ -0,0 +1,296 @@
+                    openFileDialog.InitialDirectory = Directory.GetCurrentDirectory(); // Must be an absolute path.
+                    openFileDialog.Filter = "liboeenclave.a|liboeenclave.a";
+                    openFileDialog.RestoreDirectory = true;
+
+                    if (openFileDialog.ShowDialog() != DialogResult.OK)
+                    {
+                        return false;
+                    }
+
+                    // Get the path of specified file.
+                    string filePath = openFileDialog.FileName;
+                    oeFolder = Path.GetFullPath(Path.Combine(filePath, ".."));
+                }
+            }
+            replacementsDictionary.Add("$OELibPath$", GetUnixPath(oeFolder));
+
+            // Now get the path to the associated TA Dev Kit, which should be under
+            // devkit or export-ta_arm64 or export-ta_arm32.
+            string taDevKitFolder = Path.Combine(oeFolder, "devkit");
+            if (!Directory.Exists(taDevKitFolder))
+            {
+                taDevKitFolder = Path.Combine(oeFolder, "export-ta_arm64");
+            }
+            if (!Directory.Exists(taDevKitFolder))
+            {
+                taDevKitFolder = Path.Combine(oeFolder, "export-ta_arm32");
+            }
+            replacementsDictionary.Add("$OETADevKitPath$", GetUnixPath(taDevKitFolder));
+
+            bool hardwareFloatSupported;
+            bool is32Bit;
+            GetConfFlags(taDevKitFolder, out hardwareFloatSupported, out is32Bit);
+
+            string opteeCompilerFlavor;
+            if (!is32Bit)
+            {
+                opteeCompilerFlavor = "aarch64-linux-gnu-";
+            }
+            else
+            {
+                if (hardwareFloatSupported)
+                {
+                    opteeCompilerFlavor = "arm-linux-gnueabihf-";
+                }
+                else
+                {
+                    opteeCompilerFlavor = "arm-linux-gnueabi-";
+                }
+            }
+            replacementsDictionary.Add("$OpteeCompilerFlavor$", opteeCompilerFlavor);
+
+            return true;
+        }
+
+        private bool SetEnclaveName(Dictionary<string, string> replacementsDictionary)
+        {
+            try
+            {
+                string safeitemname;
+                replacementsDictionary.TryGetValue("$safeitemname$", out safeitemname);
+
+                // Extract enclave name.
+                if (safeitemname.EndsWith("_host") && (safeitemname.Length > 5))
+                {
+                    string enclavename = safeitemname.Substring(0, safeitemname.Length - 5);
+
+                    // Add $enclavename$.
+                    replacementsDictionary.Add("$enclavename$", enclavename);
+                }
+
+                // Try to get enclave guid from the enclave project.
+                string enclaveguid = "FILL THIS IN";
+                string makFileName = Path.Combine(EdlLocation, "optee", "linux_gcc.mak");
```

Comment:
This no longer applies. The UUID of the enclave for OP-TEE is now set via a macro in OE.

Reply:
Updated to get the enclave project's guid from the enclave project's vcxproj file instead